src/solver.f90

```diff
@@ -105,18 +108,30 @@ function init(backend, mesh, time_integrator, host_allocator, &
     solver%time_integrator => time_integrator
     solver%host_allocator => host_allocator
 
-    solver%xdirps => xdirps
-    solver%ydirps => ydirps
-    solver%zdirps => zdirps
+    allocate (solver%xdirps, solver%ydirps, solver%zdirps)
+    solver%xdirps%dir = DIR_X
+    solver%ydirps%dir = DIR_Y
+    solver%zdirps%dir = DIR_Z
 
     solver%u => solver%backend%allocator%get_block(DIR_X, VERT)
     solver%v => solver%backend%allocator%get_block(DIR_X, VERT)
     solver%w => solver%backend%allocator%get_block(DIR_X, VERT)
 
-    solver%dt = globs%dt
-    solver%backend%nu = globs%nu
-    solver%n_iters = globs%n_iters
-    solver%n_output = globs%n_output
+    ! set defaults
+    Re = 1600._dp; dt = 0.001_dp; n_iters = 20000; n_output = 100
+    poisson_solver_type = 'FFT'
+
+    if (command_argument_count() >= 1) then
+      call get_command_argument(1, input_file)
+      open (100, file=input_file)
+      read (100, nml=solver_params)
+      close (100)
+    end if
```

Comment:
Any thoughts on getting a command argument inside a subroutine like this?

Reply:
I'm not sure what you mean by the question, I think you need to get the count (as done) and then index into this to find the value you want (as done).

Reply:
I was thinking maybe passing the file name as an argument when instantiating a solver object from the main program. Because otherwise if a main program uses solver class it requires an argument, but if the file name is passed as an argument the main program can be more flexible when retrieving and passing the input file.